scripts/solve_network.py

```diff
@@ -167,6 +168,28 @@ def add_CCL_constraints(n, config):
                                            '<=', maximum, 'agg_p_nom', 'max')
 
 
+def add_EQ_constraints(n, o, scaling=1e-1):
+    float_regex = "[0-9]*\.?[0-9]+"
+    level = float(re.findall(float_regex, o)[0])
+    if o[-1] == 'c':
+        ggrouper = n.generators.bus.map(n.buses.country)
+        lgrouper = n.loads.bus.map(n.buses.country)
+        sgrouper = n.storage_units.bus.map(n.buses.country)
+    else:
+        ggrouper = n.generators.bus
+        lgrouper = n.loads.bus
+        sgrouper = n.storage_units.bus
+    load = n.snapshot_weightings @ \
+           n.loads_t.p_set.groupby(lgrouper, axis=1).sum()
+    inflow = n.storage_units_t.inflow.groupby(sgrouper, axis=1).sum().sum()
+    inflow = inflow.reindex(load.index).fillna(0.)
+    rhs = scaling * level * ( load - inflow )
+    lhs = linexpr((n.snapshot_weightings * scaling,
+                   get_var(n, "Generator", "p").T)
+                 ).T.groupby(ggrouper, axis=1).apply(join_exprs)
+    define_constraints(n, lhs, ">=", rhs, "equity", "min")
+
```

Comment:
Nice syntax here! 

However here is a problem with the (load - inflow) part. Note you still have the spillage variable, i.e. not all of the inflow is converted into power production. I see two options, either add the negative spillage variable to the lhs (then the inflow should be outside of the parantheses), or add the positive power production of storage units to the lhs. 

Here the equations

$p_{generators}^{c} + p_{storage}^d = \chi d^c$

where c is the group, $\chi$ the level and $p^c$ the total power production per group. With 

$p_{storage}^c = p_{inflow}^c - p_{spillage}^c$

you see what I mean.

Reply:
Sorry, it does not render the latex equations...

Reply:
> Note you still have the spillage variable

Ah, yes! I'll see how to implement it.

> Sorry, it does not render the latex equations...

I wonder why Github does not manage to add this feature...

Reply:
Two options then:
1) weight * (generation - spill) + inflow >= level * weight * load
2) generation + storage discharge - storage charge >= level * load
I have a slight preference for 1) because fewer variables. @FabianHofmann what do you think?